plugins/MachineSettingsAction/MachineSettingsAction.qml

```diff
@@ -616,7 +621,14 @@ Cura.MachineAction
                         {
                             if (propertyProvider && text != propertyProvider.properties.value)
                             {
-                                propertyProvider.setPropertyValue("value", text);
+                                if (_setValueFunction !== undefined)
+                                {
+                                    _setValueFunction(text);
+                                }
+                                else
+                                {
+                                    propertyProvider.setPropertyValue("value", text);
```

Comment:
I don't really understand why this is necessary. Could you document why it's necessary to change the value of the materialDiameterField via this _setValueFunction? Shouldn't we just be listening to the value change on the other end? Update the field on the activeStack that way?

Reply:
Documented in ea9aa2c77